tests/js/articles.frontend.spec.js

```diff
@@ -0,0 +1,139 @@
+import {
+  createNewPost,
+  enablePageDialogAccept,
+  setPostContent,
+  publishPost,
+} from '@wordpress/e2e-test-utils';
+
+import {
+  selectBlockByName,
+  openSidebarPanelWithTitle,
+  clickElementByText,
+  typeInDropdownWithLabel,
+} from './e2e-tests-helpers';
+
+import 'expect-puppeteer';
+
+const ARTICLS_BLOCK = `
+<!-- wp:planet4-blocks/articles {"article_count":2} -->
+<div class="wp-block-planet4-blocks-articles" data-render="planet4-blocks/articles" data-attributes="{&quot;attributes&quot;:{&quot;article_heading&quot;:&quot;Related Articles&quot;,&quot;article_count&quot;:2,&quot;tags&quot;:[],&quot;posts&quot;:[],&quot;post_types&quot;:[],&quot;read_more_text&quot;:&quot;Load More&quot;,&quot;read_more_link&quot;:&quot;&quot;,&quot;button_link_new_tab&quot;:false,&quot;ignore_categories&quot;:false},&quot;innerBlocks&quot;:[]}"></div>
+<!-- /wp:planet4-blocks/articles -->
+`;
+
+describe( 'Articles block frontend', () => {
+  beforeAll( async () => {
+    // This helps in overriding Wordpress dialogs preventing actions.
+    await enablePageDialogAccept();
+  } );
+
+  it ( 'test post thumbnail behaviour', async () => {
+    // Create new post.
+    await createNewPost({
+      title: 'Test Articles with no thumbnail',
+      content: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.',
+      showWelcomeGuide: false,
+    });
+    await publishPost();
+    await page.waitForSelector( '.post-publish-panel__postpublish' );
+
+    //Create a new page and add newly created post in it.
+    await createNewPost({
+      postType: 'page',
+      title: 'Test Articles block thumbnail',
+      showWelcomeGuide: false,
+    });
+
+    // Add articles block in page.
+    await setPostContent( ARTICLS_BLOCK );
+
+    await selectBlockByName( 'planet4-blocks/articles' );
+
+    await openSidebarPanelWithTitle( 'Setting' );
+
+    // Manually add post(without thumbnail img) in articles blocks.
+    await typeInDropdownWithLabel( 'Manual override', 'Test Articles with no thumbnail' );
+
+    // Manually add post(post with image inside post content, which was set as a feature image
+    // & display as a thumbnail in articles block) in articles blocks.
+    await typeInDropdownWithLabel( 'Manual override', 'Lorem ipsum dolor sit amet' );
+
+    // Publish the page.
+    await publishPost();
+
+    await new Promise((r) => setTimeout(r, 500));
+
+    await page.waitForSelector( '.post-publish-panel__postpublish' );
```

Comment:
The steps in [this part](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/354/files#diff-681f3eaecf153bfb54d00857cbae7c41R40-R65) need to be executed one by one in the editor. If these functions are already tested in the editor test, then we don't need to execute them here and can just insert the corresponding block comment into the post. That could make the test a lot faster.

Reply:
You might need to get the post id for the post that is created [here](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/354/files#diff-681f3eaecf153bfb54d00857cbae7c41R40) in order to to that.

Reply:
Thats my initial thought to get post_id of new post and add it directly in block comment and use it, but somehow I didn't find a way to get post_id that way... :(